### PR TITLE
docs(eslint-plugin): [ban-types] add reference on how to type an empty object

### PR DESCRIPTION
**Pull request by martinfrancois**
I recently got a `ban-types` error when specifying an empty object to be allowed (`{}`), which correctly warned me that this is not what I intended. But the message only offered alternatives for cases that didn't apply to my case in particular. After a lot of digging I found this issue (https://github.com/typescript-eslint/typescript-eslint/issues/2063) thankfully explaining the solution to my problem.

I want to save some other devs the trouble of having to look for a solution in this case (or even worse, some people probably ignoring the error, should they not find a solution). This was also suggested here https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-659748424

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/ban-types.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ The default options provide a set of "best practices", intended to provide safet
   - It accepts class declarations, which will fail when called, as they are called without the `new` keyword.
 - Avoid the `Object` and `{}` types, as they mean "any non-nullish value".
   - This is a point of confusion for many developers, who think it means "any object type".
+  - See [microsoft/TypeScript#2063](https://github.com/typescript-eslint/typescript-eslint/issues/2063).
```
Comment (bradzacher) [inline packages/eslint-plugin/docs/rules/ban-types.md:77]: microsoft/TypeScript#2063
```suggestion
  - See [this comment for more information](https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492).
```

File: packages/eslint-plugin/src/rules/ban-types.ts [hunk 2]
```diff
@@ -93,6 +93,9 @@ const defaultTypes: Types = {
       '`{}` actually means "any non-nullish value".',
       '- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.',
       '- If you want a type meaning "any value", you probably want `unknown` instead.',
+      '- If you want a type meaning "an empty object", you probably want to use the following instead:',
+      '  type NoElements<T> = { [P in keyof T]: never };',
+      '  type EmptyObject = NoElements<Record<string, never>>;',
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/ban-types.ts:98]:
```suggestion
      '- If you want a type meaning "empty object", you probably want `Record<string, never>` instead.',
```

**Review by bradzacher — CHANGES_REQUESTED** [review]

**Comment by martinfrancois** [comment]
@bradzacher thanks for the suggestions! I applied them.

**Review by bradzacher — APPROVED** [review]
thanks!

**Pull request merged**